pandas/core/generic.py

```diff
@@ -5438,6 +5438,9 @@ def _get_numeric_data(self):
     def _get_bool_data(self):
         return self._constructor(self._mgr.get_bool_data()).__finalize__(self)
 
+    def _get_numeric_or_datetime_data(self):
+        return self._constructor(self._mgr.get_numeric_or_datetime_data()).__finalize__(self)
```

Comment:
let's instead of adding to the internal routines, let's use `.select_dtypes()` here. I think more idiomatic (we should also change the others).

Reply:
Refactored, I think the updated version is clearer.